### PR TITLE
FlatGeobuf: Port upstream changes to fix oss-fuzz 27530

### PR DESCRIPTION
**Pull request by bjornharrtell**

### CODE REVIEW

File: gdal/ogr/ogrsf_frmts/flatgeobuf/packedrtree.cpp [hunk 1]
```diff
@@ -211,10 +222,8 @@ std::vector<std::pair<uint64_t, uint64_t>> PackedRTree::generateLevelBounds(cons
     // bounds per level in reversed storage order (top-down)
     std::vector<uint64_t> levelOffsets;
     n = numNodes;
-    for (auto size : levelNumNodes) {
-        levelOffsets.push_back(n - size);
-        n -= size;
-    }
+    for (auto size : levelNumNodes)
+        levelOffsets.push_back(n -= size);
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/flatgeobuf/packedrtree.cpp:226]: can you move the n-= size; in the line before for better code clarity ?

Comment (bjornharrtell) [reply gdal/ogr/ogrsf_frmts/flatgeobuf/packedrtree.cpp:226]: unless you feel very strongly about it I would prefer to keep it this way

Comment (rouault) [reply gdal/ogr/ogrsf_frmts/flatgeobuf/packedrtree.cpp:226]: as you wish

**Comment by bjornharrtell** [comment]
@rouault I've searched a few of the CI build failures but they make no sense to me.

**Comment by bjornharrtell** [comment]
Think I've found it now:

2020-12-26T16:12:51.8133544Z [91mpackedrtree.cpp: In function 'uint32_t FlatGeobuf::hilbert(const FlatGeobuf::NodeItem&, uint32_t, double, double, double, double, double, double)':
2020-12-26T16:12:51.8136753Z packedrtree.cpp:130:109: error: unused parameter 'maxX' [-Werror=unused-parameter]
2020-12-26T16:12:51.8139163Z   130 | uint32_t hilbert(const NodeItem &r, uint32_t hilbertMax, const double minX, const double minY, const double maxX, const double maxY, const double width, const double height)
2020-12-26T16:12:51.8140199Z       |                                                                                                ~~~~~~~~~~~~~^~~~
2020-12-26T16:12:51.8143248Z packedrtree.cpp:130:128: error: unused parameter 'maxY' [-Werror=unused-parameter]
2020-12-26T16:12:51.8144646Z   130 | uint32_t hilbert(const NodeItem &r, uint32_t hilbertMax, const double minX, const double minY, const double maxX, const double maxY, const double width, const double height)

**Comment by bjornharrtell** [comment]
@rouault this should now be ready to merge.

**Pull request merged**